### PR TITLE
feat: consent-ledger echo + re-consent click-through (tos-v1 phase 2)

### PR DESCRIPTION
**Pull request by jjramirezn**
## Summary

Client half of the consent ledger (peanut-api #1209, tos-v1 phase 2): every acceptance surface now echoes the exact legal-document revisions it displayed, and the ToS §17 re-consent click-through ships.

- **Generated versions:** `scripts/generate-legal-versions.mjs` (predev + prebuild, output committed) derives `slug → { version, hash }` from the `src/content` legal frontmatter (`last_updated` + sha256 of each `en.md`). No hardcoded versions; bumping the content submodule refreshes the constants on next build.
- **Signup echo:** the ZeroDev SDK owns the `/passkeys/register/verify` body, so `useZeroDev.handleRegister` sends displayed terms+privacy versions via the `x-accepted-legal` header (`passkeyServerHeaders`).
- **Card apply echo:** `rainApi.applyForCard` gains `acceptedDocuments`; the card page sends the exact set CardTermsScreen showed (US: card-terms-us + card-esign + card-privacy; international: card-terms-international + card-esign).
- **Re-consent:** new `ReConsentModal` (mobile-ui layout) checks `GET /users/consent/status` once per session; when a published version passed the user's last provably-accepted one, a blocking checkbox click-through lists the updated docs and records acceptance via `POST /users/consent/accept`.
- New client-side service `src/services/consent.ts` (no server actions).

## Risks / breaking changes

- **Cross-repo:** pairs with peanut-api #1209. Either merge order is safe — the echoes are optional fields/headers the old backend ignores, and a 404 on consent-status just logs (fail-open, app never locks).
- **All pre-ledger users see the re-consent modal once** after the backend ships — by design (nobody has provable consent; accepting backfills the ledger). If product wants a quieter rollout, gate the modal mount.
- The modal is deliberately unclosable until accepted (ToS §17); a failed status fetch keeps the app usable.

## QA

1. `pnpm dev` → predev regenerates `src/constants/legal-versions.generated.ts` (7 docs).
2. Fresh signup → backend rows for `terms` + `privacy` @ `2026-07-15` with the generated hashes.
3. Card apply (US + international) → regional doc-set rows, `surface='card-apply'`.
4. Pre-ledger user (or backdated row) → modal on app load; doc links open (web: new tab, native: in-app browser); accept → `surface='re-consent'` rows; modal never returns.
5. Backend down → app loads normally, console warning only.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Design notes / accepted trade-offs

- **Generated constants are committed** so CI and static analysis see them without running the generator; predev/prebuild regenerate, and the output is byte-stable under prettier by construction.
- **Version skew self-heals:** if content ships before this app redeploys, a re-consent acceptance records the older displayed version (backend drift-warns), the modal reappears next session, and the next FE deploy trues it up.
- **Reveals, not blocking:** legal-doc links now live in two maps (`CardTermsScreen`'s `LINKS`, absolute URLs for native, and `ReConsentModal`'s `DOC_LABELS`, relative for DocsLink). Unifying them into one legal-links const is a small follow-up.

## Summary by CodeRabbit

* **New Features**
  * Added prompts for existing users to review and accept updated legal documents before continuing.
  * Added clear document links, acceptance checkbox, loading states, and error handling.
  * Legal document consent is now recorded during account registration and card applications.
* **Bug Fixes**
  * Ensured consent reflects the user’s residency when applying for a card.
* **Analytics**
  * Added tracking for re-consent modal display and completion actions.

### CODE REVIEW

**Comment by jjramirezn** [comment]
@coderabbitai review

**Comment by jjramirezn** [comment]
@coderabbitai review

**Comment by jjramirezn** [comment]
@coderabbitai review

**Comment by jjramirezn** [comment]
@coderabbitai full review

File: package.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
         "dev:fallback": "next dev",
         "postinstall": "node scripts/copy-flags.mjs",
         "copy-flags": "node scripts/copy-flags.mjs",
-        "prebuild": "node scripts/copy-flags.mjs",
+        "predev": "node scripts/generate-legal-versions.mjs",
+        "prebuild": "node scripts/copy-flags.mjs && node scripts/generate-legal-versions.mjs",
```
[the remark this answers is not in the document]
Comment (jjramirezn) [reply package.json]: Fixed in be5545bfa — extracted generate:legal-versions as a named script and wired pre-hooks for every entrypoint that renders content: dev, dev:clean, dev:fallback, analyze, and build.

File: src/components/Global/ReConsentModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        if (!userId || lastCheckedUserId.current === userId) return
+        lastCheckedUserId.current = userId
+        consentApi
+            .getStatus()
+            .then((status) => {
+                if (!status.needsReConsent) return
+                // only prompt for documents this client can actually display
+                const docs = status.documents.filter((d) => d.needsAcceptance && d.slug in LEGAL_DOCUMENT_VERSIONS)
+                if (!docs.length) return
+                setOutdatedDocs(docs)
```
[the remark this answers is not in the document]
Comment (jjramirezn) [reply src/components/Global/ReConsentModal/index.tsx:58]: Fixed in be5545bfa — on account switch the modal clears docs/checkbox/error before fetching, a status response for the previous user is discarded (userId guard against lastCheckedUserId at resolve time), and the checkbox resets after a successful acceptance so a later appearance starts unticked.

**Review by Hugo0 — APPROVED** [review]
Approving — lean and internally consistent: one service file, one modal, one build-time generator, no speculative abstractions. Backend-decided `needsAcceptance` (no client date math), fail-open status fetch, fail-closed acceptance is the right split. Verified the committed `legal-versions.generated.ts` is byte-identical to the content submodule at the pinned gitlink, and the card doc sets exactly match what `CardTermsScreen` actually displays. Wire contract checked field-by-field against peanut-api #1209 — matches.

Asks before/alongside merge:

1. **Add tests — this modal gates the entire app and has zero.** Minimum set (patterns in-tree, see `StaleCardApproval/__tests__/ReEnableModal.test.tsx`):
   - fail-open: status-fetch failure must NOT show/lock the modal (the load-bearing safety property, currently unasserted)
   - accept failure keeps modal + shows retry error
   - account-switch reset (the regression CodeRabbit already caught once)
   - `cardConsentDocuments(true/false)` doc sets — silent drift here mis-ledgers legal consent, the exact bug class this PR exists to prevent
2. **Swap the two `console.error`s for Sentry captures.** If `POST /accept` 500s systematically, every user is stuck behind an undismissable retry modal and prod would be silent about it.
3. **Product sign-off on the big-bang wave**: every existing user gets the modal once after the backend ships (deliberate — it backfills the ledger). Fine legally, but someone should consciously OK the UX hit, or gate the mount for a quieter rollout.
4. **CORS cross-check with #1209**: `x-accepted-legal` must be in the API's `Access-Control-Allow-Headers` before native release — on Capacitor the passkey call is cross-origin and a rejected preflight fails the whole signup, not just the header.

Notes (non-blocking):
- Known nag-loop if content ships before the FE redeploys (accepting records the older version → modal returns every session until deploy). Acknowledged in the PR body; keep deploy order in mind.
- Modal can pop mid-flow (mounted in layout, shows when the async fetch resolves) — low probability, worth knowing.
- Generator's `last_updated` regex is m-flagged over the whole file; anchoring it to the frontmatter block is a one-line tightening.
- eslint failure is base debt (only touched-file hit is pre-existing `useZeroDev.ts:161`); required `ci-success` is green.

**Comment by jjramirezn** [comment]
Review asks addressed (@Hugo0) in f06fd719:

**1. Tests** — `ReConsentModal/__tests__/index.test.tsx` (7 tests, `ReEnableModal.test.tsx` pattern):
- fail-open: a failed status check never shows/locks the modal (the load-bearing safety property)
- accept failure keeps the modal, shows the error, leaves retry enabled
- account-switch: stale status response from the previous user is discarded; state resets per user, one check per user per session
- undisplayable-slug filtering (docs this build doesn't know never trigger the modal)

Plus `src/services/__tests__/consent.test.ts` pinning the exact `signupConsentDocuments` / `cardConsentDocuments(true|false)` slug sets and that every entry carries its own generated version+hash — the silent-drift-mis-ledgers-consent class.

**2. Sentry** — both `console.error`s are now `Sentry.captureException` with `feature: 're-consent'` tags (`action: 'status'` / `'accept'`), so a systemic `/accept` failure trapping users behind the modal is visible in prod.

**3. Big-bang wave** — product call is yours; flagging that it stands: every existing user sees the modal once after the backend deploys.

**4. CORS** — verified on the API side (no change needed): our `@fastify/cors` config sets no `allowedHeaders`, so it reflects the preflight's requested headers for permitted origins (incl. `capacitor://`); the native app's existing `x-username` header already proves the path. Details on peanutprotocol/peanut-api-ts#1209.

Erasure runbook (the API-side ask): https://github.com/peanutprotocol/mono/pull/53.

**Comment by jjramirezn** [comment]
Closing: content fully landed on dev — generate-legal-versions.mjs and src/services/consent.ts are on dev, diff vs dev empty.
